js/lib/DateFactory.js

```diff
@@ -259,4 +259,19 @@ DateFactory._dateToIlib = function(inDate, timezone, locale) {
     });
 };
 
+DateFactory._ilibToDate = function(ilibDate, timezone, locale) {
+    if (typeof(ilibDate) === 'undefined' || ilibDate === null) {
+        return ilibDate;
+    }
+    var year = ilibDate.year;
+    var month = ilibDate.month-1;
+    var day = ilibDate.day;
+    var hours = ilibDate.hour;
```

Comment:
`new Date()` uses the local time zone offset only. So if the ilibDate does not use the "local" time zone and it does not use the current time zone, then the above doesn't work. We can always get the right time by doing this:

```
return new Date(ilibDate.getTimeExtended());
```

`ilibDate.getTimeExtended()` returns the UTC-based unix time, and `new Date` accepts a single parameter like that. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date